openedx/core/djangoapps/catalog/utils.py

```diff
@@ -75,21 +79,23 @@ def check_catalog_integration_and_get_user(error_message_field):
         return None, catalog_integration
 
 
-def get_programs(site, uuid=None):
+def get_programs(site=None, uuid=None, course=None):  # pylint: disable=redefined-outer-name
     """Read programs from the cache.
 
     The cache is populated by a management command, cache_programs.
 
-    Arguments:
-        site (Site): django.contrib.sites.models object
-
     Keyword Arguments:
+        site (Site): django.contrib.sites.models object
         uuid (string): UUID identifying a specific program to read from the cache.
+        course (string): course id identifying a specific course run to read from the cache.
 
     Returns:
         list of dict, representing programs.
         dict, if a specific program is requested.
     """
+    if len([arg for arg in (site, uuid, course) if arg is not None]) != 1:
+        raise TypeError('get_programs takes exactly one argument')
```

Comment:
Minor, but I'd second @BessieSteinberg's suggestion from the prior PR

"I would include in the docstring that site, uuid OR course should be set, but only one"